File: Assets/scripts/ground_detection.js
#pragma strict

//lives variable to be read by the canvas to update the scoreboard
public var livesLeft = 3;

function Start () {
	
}

function Update () {
	
}

function OnCollisionEnter2D (coll: Collision2D) {
    //if the ground collides with a created bomb
    if(coll.gameObject.name == "bomb_Prefab(Clone)") {
        Destroy(coll.gameObject);
        livesLeft--;
        
        //Identify all bombs & destroy them
        var getAllBombs = GameObject.FindGameObjectsWithTag("bomb");
    	for (var i = 0; i < getAllBombs.length; i++) {
    		Destroy(getAllBombs[i]);	
    	}
    }
}

File: Assets/scripts/ground_detection.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ground_detection : MonoBehaviour {

	void OnCollisionEnter2D (Collision2D coll) {
		//if the ground collides with a created bomb
		if(coll.gameObject.name == "bomb_Prefab(Clone)") {
			//Destroy the bomb that collided
			Destroy(coll.gameObject);
			
			//Stop dropping bombs
			GameManager.instance.bombsAreBeingDropped = false;

			//Identify all bombs on screen & destroy them
			var getAllBombs = GameObject.FindGameObjectsWithTag("bomb_prefab");
			for (var i = 0; i < getAllBombs.Length; i++) {
				Destroy(getAllBombs[i]);	
			}

			//Decrement the lives counter
			GameManager.instance.livesLeft--;
		}
	}
}
